### DISCUSSION TITLE
SideDownload Error and Exit Code:1 Error

### DISCUSSION THREAD

**Discussion by jazzinvicki** [aclap-dev/video-downloadhelper#162] [Help and Q&A]
### Your question

THIS IS MY INITIAL QUESTION SENT VIA EMAIL:
After the recent update, I can no longer get it to work. I just now purchased another license... A Conversion License for Firefox, thinking this was my issue, but I am still having this error:

SideDownload error: 
[dash @ 0x7fcdaca0a380] Error when loading first fragment of playlist
[in#0 @ 0x7fcdaca0a240] Error opening input: Server returned 400 Bad Request
Error opening input file https://29vod-adaptive.akamaized.net/exp=1708275750~acl=%2F05982558-291f-44b5-a547-166d8b042f44%2F%2A~hmac=6ae22b389cdee420c6a861677c64ff4242b61d2a6d5c57c6fe123e279020287b/05982558-291f-44b5-a547-166d8b042f44/sep/video/3b1aa1c9,4097daa6,61dee2c1,63632818,ebd46b9c/audio/9f0d155e,9fc0a962,e3520f20/master.mpd?base64_init=1&query_string_ranges=1.
Error opening input files: Server returned 400 Bad Request

2ND EMAIL UPDATE: 
I just noticed a new post on the help site which showed a Beta version to install for now.  I did that.  I thought for a second it would work and then got a different error:

Exit code: 1

moz-extension://a19f8c2e-8710-4df0-beb9-272937d0019c/background/main.js:1
receive@moz-extension://a19f8c2e-8710-4df0-beb9-272937d0019c/background/main.js:1:13157 
Zp</callCatchAppNotFound/</<@moz-extension://a19f8c2e-8710-4df0-beb9-272937d0019c/background/main.js:1:42624

**Comment by jazzinvicki** [comment]
Just to add to my earlier post. -
I normally go to our corporate website and play the videos I want to save. Click Download and Convert to MP4-2 as I have for many years now and save to my hard drive. Since your update this no longer works. I’ve paid for a license twice now. Any help appreciated. Thanks for all you do.

**Comment by jazzinvicki** [comment]
Fixed!  I deleted the new extension. Downloaded 8.1.4.0 and installed it.  Works like new.

**Reply by Niazi78675** [reply]
From Where can I get this version?

**Reply by jazzinvicki** [reply]
I just googled it until I found it. I don’t remember exactly. It was easy to find.

**Reply by Niazi78675** [reply]
i am trying but didn't find that version. help could you please help with the link of that exact version

**Reply by GrampaWildWilly** [reply]
Go to addons.mozilla.org.
Enter `video` into the "Find add-ons" search field within that web page.  It's in the upper right area of the page.
The first proposed entry in the autocomplete dropdown is for VDH.
Click that.
That takes you to the VDH install page.
Look for this on that page:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/be96a950-9421-4626-9a8d-aca7a3c749a7)

Click that link & you will find an archive of a small number of backlevel versions of VDH.

People need to understand that the VDH folks don't actually update VDH on your computer.  The browser does that.  It does that under control of a setting that you control.  You tell your browser whether to automatically update the extension when a new version has been released.  I strongly recommend you turn off automatic updates for VDH.  Beyond the reason that probably most appeals to you, that of being able to stick on a backlevel version, there is another reason that I consider to be really insidious.  If you happen to be downloading something with VDH & the browser -- I want to emphasize, it is the BROWSER that makes this decision, not the VDH software, not the VDH techs, the BROWSER -- if the browser decides it's time for the automatic update, the browser proceeds with the automatic update to VDH.  This kills whatever downloads you might have in progress.  A few user reports of this actually happening were posted on our old Google Groups forum.  Other addons might not  have this as a consideration.  But VDH sure as shooting does.  So you should disable automatic updates to VDH on principle.

It is my understanding that @paulrouget is working on fixing this automatic update issue.  But the timeline for that appears to be rather long.  So don't hold your breath.  Turn off automatic updates for VDH.  When you do that, a new button appears in the browser settings for manually checking for updates.  If it tells you an update is available, you then have the option of installing the update or ignoring it.  But this is all in the browser settings.  Not the VDH settings.  The browser seettings.

**Reply by Niazi78675** [reply]
Yeah Thank You I found the desired version but my video is still getting error. 
my URL: https://asmr.vids.io/videos/1190d7b11310e4ce98/welcome-massage-by-anna-1080p

**Reply by Niazi78675** [reply]
Could You Please help me with the video?

**Reply by GrampaWildWilly** [reply]
When I go to that page, it shows me this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/0cafda14-ef4e-43c1-b50a-7b321296ada2)

I tried to download the first 2 variants & they both gave errors.  The third one downloaded something but it was only 8 seconds long.

Time for Plan B.

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/8e01ee11-10d2-4d61-b922-f4d524e7c620)

Unfortunately, ffprobe on that HLS master manifest gave 403 Forbidden.  There appears to be some sort of stealth DRM in place here.  This one has me blocked.  Maybe somebody else can figure it out.

**Reply by Mattmjs** [reply]
> Yeah Thank You I found the desired version but my video is still getting error.
my URL: https://asmr.vids.io/videos/1190d7b11310e4ce98/welcome-massage-by-anna-1080p

Downloading will not be possible on this one, each stream manifest is paired with a key that I see in the developer tools.
I assume this is an encryption key, so no matter what is tried it will encounter some error. Kind of like DRM.

**Reply by Niazi78675** [reply]
Yeah I also tried different versions on this video and each version is giving some sort of error I Also thought maybe it is a DRM protected video but question here is does't VDH download DRM videos? I think it should have

**Reply by GrampaWildWilly** [reply]
The mere presence of key information in a stream manifest does not necessarily mean it's unavailable for free download.  All the Medici TV content I have ever downloaded has AES-128 key information, but it downloads in VDH or directly with ffmpeg no problem.  The Network Monitor for this massage video appears to show the 2 key files 240.key & 1080.key.  But the manifests don't seem to have the complete URLs of the key files.  The manifests just say 240.key & 1080.key but the URLs of the 2 key files are .key?Policy= and a bunch of unintelligible gobbledygook.  I'm guessing they're the key files.  Their Type under the Type column is application/octet-stream.  It looks to me like the stream manifests are intentionally incorrectly built.  This amounts to stealth DRM.  But the player in the page has no problem with it, probably because it can read the application/octet-stream object that is supplied to the client browser.  We're blocked because we can't use the manifests they've supplied.  It's possible to save the manifests to your system so I suppose it would be possible to hack in the full URL of a key file.  But the chunk descriptors making up the rest of the stream manifest are partial URLs.  You would need to hack in the full URL names for each one of those.  I would imagine that's possible because while playback is in progress in the web page, the Network Monitor shows the mp2t chunks.  You could grab the URL of one of those & put together the full URLs.  You would have to supply full URLs because if a manifest is on your system, ffmpeg thinks your system is the host of the content & doesn't generate the correct requests, which have to be directed at the real web site where the content really is hosted.  I'm not sufficiently motivated to do that much fiddling, especially since I am guessing.  As usual.

**Reply by GrampaWildWilly** [reply]
VDH has never, does not now, & will never download content protected by DRM.  Paul posted a comment to that effect recently.  Digital Rights Management is intended to block downloading of copyrighted material.  VDH does not do whatever hacking might be necessary to break DRM protection.  And it should not.  If people want to protect their copyrighted material, they are entitled to do so.  However, when I disabled DRM support in Firefox before visiting this page, it did not pop up the warning that the page contained DRM-protected content.  It just played the video despite my having disabled the browser's DRM support.  So this is not an issue of actual DRM protection.  That's why I'm calling this stealth DRM.  It's not really DRM but it has the effect of DRM.

**Reply by paulrouget** [reply]
I found out what the issue is: non-standard segment & key retrieval mechanism.

Technical explanation: HTTP parameters are not carried from M3U8 URL parameters to key and segments URLs. Usually this is done via Cookies, but in that case, it's done via URL parameters. This was supported in older version of VDH, but not anymore via ffmpeg. That's something I'd like to fix one day.

Sadly. This won't be fixed in the coming days. Need more time to figure that one out.

**Reply by Niazi78675** [reply]
So Now Is it still supported in Older versions or not?

**Reply by paulrouget** [reply]
Yes. I recommend you install an older version from https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper/versions/

**Reply by Niazi78675** [reply]
But I tried the previous two versions as recommended by @jazzinvicki too but they gave a different sort of error (each version)

**Reply by GrampaWildWilly** [reply]
I suspect that it would be enough to use an older version of the CoApp.  CoApp versions 2.x use ffmpeg.  Older ones do not.  But I don't know if CoApp 1.6.3 for example would work with any VDH version 8.x.  The last release of VDH that I ran with CoApp 1.6.3 was a 7.something, which of course I have totally forgotten now.  You might grab one of the 7.x beta releases from the old CoApp repository here: https://www.downloadhelper.net/firefox/betas.

Honestly, I think that would be a terribly drastic measure just to be able to download this video.

So Paul.  I'm reading your explanation about non-standard segment & key retrieval as being a more technically accurate wording of my long-winded rambling about completing URLs.  Am I correct?  I'm referring to my post above that begins, "The mere presence of key information in a stream manifest . . ."

**Reply by GrampaWildWilly** [reply]
Well, that got me sufficiently motivated.  I went back to the massage web page & saved the 1080 stream manifest on my system.  That was by popping up the context menu on that application/x-mpegurl object in the Network Monitor & executing Copy Response on the cascaded submenu.  Then I got the URL of the 1080.key file & pasted it somewhere I could access it.  Then I hunted for an mp2t file in the Network Monitor & pasted its URL somewhere I could access it.  I then used those 2 pastes to flesh out the parts missing from the manifest & saved these modifications on my system.

[Massagem3u8.txt](https://github.com/aclap-dev/video-downloadhelper/files/14358199/Massagem3u8.txt)

I had to change the file name by tacking on .txt to satisfy GitHub.  But the real name of the file was Massage.m3u8 when I pushed that through ffmpeg.

[Massage Video mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14358126/Massage.Video.mp4.Log.txt)

Holy cow . . .

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/a16176cb-077f-4e2c-9bb4-01ca6b08c038)

Playing it in VLC went just fine.  There isn't much of a sound track to speak of.  The 2 ladies don't say a single word in this thing.  (I just sampled it to make sure it was all there.  It's not interesting enough for me to actually sit & watch this.)  The name of the web site does include the qualifier ASMR so you would expect whatever audio there is to be nearly inaudible.  I could only just barely hear anything at all even with my volume cranked up to what is normally ear-splitting.  The only audio here was the sound of the masseuses hands rubbing over the client's skin, & that rose to the level of audibility only intermittently.

I was intrigued by the things the masseuse was wearing on her wrists.  It took a while to find a place in the video that got a clear enough closeup of the one on her left wrist.

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/d447bbb8-3c86-45f4-9e37-91d1ac98535f)

A closer look.

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/a0084516-d134-41d4-a914-ffada1198807)

A quick web search got me here: https://rode.com/en-us/microphones/wireless/wirelessgoii.  I'm pretty sure that's what she's wearing on her wrists to pick up the binaural ASMR audio.

Anyway, it seems I've answered my own question.  I was 100% correct above.  This is the task ahead of Paul.  I don't envy you, mon pote.  This indeed does not look simple.

**Reply by Niazi78675** [reply]
Man You are literally a genius So now i have to try this all

**Reply by paulrouget** [reply]
@GrampaWildWilly yep - we have some other similar issues with some other websites. Adding the parameters manually will indeed do the trick.

The issue is that we can't do it for all videos as most of the time it will break other downloads. It's only some websites that use that non-standard behavior.

I was thinking of just trying twice: standard way, and if fail, non standard way.

**Reply by GrampaWildWilly** [reply]
@Niazi78675 Not a genius.  Just a determined geek.

@paulrouget Is the non-standard way something that is likely to apply on a decent subset of web sites?  I'm looking at this example & I'm thinking every site would need its own hand-crafted solution.  Which is to say, writing the manual process into computer code is not on the books.  I had to add both a prefix & a suffix to each partial URL for each chunk in the media manifest.  I believe I've seen other sites on which only a prefix would have been necessary.  There is such chaos around the web when it comes to the presentation of audio-visual content.  There is no enforceable ISO standard for it.  I've said it many times before: It's something of a miracle that VDH works on as many web sites as it does.

**Comment by paulrouget** [comment]
I need the URL of the video you're trying to download. Otherwise I can't help.

**Comment by jazzinvicki** [comment]
The URL is not public. It’s for Jazzercise instructors only and an
instructor ID and password is needed. But here it is…

https://thestudio.wisetail.com/eco_login.php

I deleted the updated extension from Firefox and reinstalled the previous
version for now until you get things worked out.

Thank you. Vicki

**Reply by paulrouget** [reply]
I tried the video that's available here: https://ondemand.jazzercise.com/ and it seems to work.

> Addon & CoApp version and details
> No response

You didn't mention any details here. I need this to assist you: https://github.com/aclap-dev/video-downloadhelper/wiki/Where-can-I-find-my-versions%3F

And please make sure the latest coapp is installed: https://www.downloadhelper.net/w/CoApp-Installation

**Comment by jazzinvicki** [comment]
So did you want me to try another website or video? This isn’t the video I
need. You asked for my URL.  receive about 15 videos every month that I
download to my dropbox. I don’t use it for anything else.

I did say I had the most current version you have released. I also
purchased again yesterday the most current CoApp version.

Then… I received an email from you yesterday stating this:

Your Video DownloadHelper license has been temporarily disabled due to too
many installations. This can be due to the fact you have created several
browser user profiles or you reinstalled your browser several times.

Please follow the link below to re-enable your license


I’ve used your extension for years with no issues. Only after the most
recent update did these errors occur.

I’m back to using the previous version for now. It works perfectly.

Thank you

Vicki

**Comment by jazzinvicki** [comment]
The new version updated automatically.  So it is no longer working... here
ya go...


version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): 9d80b8f7d4554b10
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:122.0) Gecko/20100101
Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.17",
  "binary": "/Applications/
net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac13",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/Users/scottkirouac",
  "converterBinary": "/Applications/
net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.1.1-tessus"
}

**Reply by paulrouget** [reply]
> The new version updated automatically

You might want to disable autoupdate in the addon settings while we are trying to figure out your issue.

**Comment by paulrouget** [comment]
> I’m back to using the previous version for now. It works perfectly.

What version is that? 8.1 or 8.0?

In version 8.2, before downloading, can you click in the submenu of the video that shows up in the popup, click details, and open the "core_media" section, and take a screenshot?

**Comment by jazzinvicki** [comment]
Ok I will. As for now I don’t need to use it again until next month. So I
am fine with leaving it as is until there is a fix.  Or whatever you need
me to do to help. Thx

Vicki

**Comment by jazzinvicki** [comment]
8.1.4.0

Vicki

**Comment by jazzinvicki** [comment]
I'm not really sure what you want a screenshot of...
[image: Screenshot 2024-02-19 at 6.36.36 AM.png]

**Reply by GrampaWildWilly** [reply]
![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/7f8e7e12-02d4-4e7d-aae6-df472a681d5b)

**Reply by paulrouget** [reply]
They are using emails to reply, not github.

**Comment by jazzinvicki** [comment]
[image: Screenshot 2024-02-19 at 6.38.47 AM.png]

**Reply by jazzinvicki** [reply]
<img width="2048" alt="Screenshot 2024-02-19 at 6 38 47 AM" src="https://github.com/aclap-dev/video-downloadhelper/assets/160416968/f92aef6a-1acd-4f57-9b7f-20105261b2b9">

**Comment by jazzinvicki** [comment]
Vicki

**Comment by Niazi78675** [comment]
> ### Your question
> 
> THIS IS MY INITIAL QUESTION SENT VIA EMAIL: After the recent update, I can no longer get it to work. I just now purchased another license... A Conversion License for Firefox, thinking this was my issue, but I am still having this error:
> 
> SideDownload error: [dash @ 0x7fcdaca0a380] Error when loading first fragment of playlist [in#0 @ 0x7fcdaca0a240] Error opening input: Server returned 400 Bad Request Error opening input file https://29vod-adaptive.akamaized.net/exp=1708275750~acl=%2F05982558-291f-44b5-a547-166d8b042f44%2F%2A~hmac=6ae22b389cdee420c6a861677c64ff4242b61d2a6d5c57c6fe123e279020287b/05982558-291f-44b5-a547-166d8b042f44/sep/video/3b1aa1c9,4097daa6,61dee2c1,63632818,ebd46b9c/audio/9f0d155e,9fc0a962,e3520f20/master.mpd?base64_init=1&query_string_ranges=1. Error opening input files: Server returned 400 Bad Request
> 
> 2ND EMAIL UPDATE: I just noticed a new post on the help site which showed a Beta version to install for now. I did that. I thought for a second it would work and then got a different error:
> 
> Exit code: 1
> 
> moz-extension://a19f8c2e-8710-4df0-beb9-272937d0019c/background/main.js:1 receive@moz-extension://a19f8c2e-8710-4df0-beb9-272937d0019c/background/main.js:1:13157 Zp</callCatchAppNotFound/</<@moz-extension://a19f8c2e-8710-4df0-beb9-272937d0019c/background/main.js:1:42624
> ### Addon & CoApp version and details
> 
> _No response_

I am also getting this exit code error 1 on firefox so how did you bypass it? by installing older version?

**Reply by jazzinvicki** [reply]
Yes. 8.1

**Reply by Niazi78675** [reply]
I TRIED this version also but again different error this time

**Reply by Niazi78675** [reply]
can you help me with the video this is 
my URL: https://asmr.vids.io/videos/1190d7b11310e4ce98/welcome-massage-by-anna-1080p

**Reply by jazzinvicki** [reply]
I’m sorry I am not home right now, so I cannot help with your videos. Maybe someone else can. Otherwise I can help tomorrow.

**Reply by Niazi78675** [reply]
Ok Bro plz give it a go when you get free i will wait

**Reply by jazzinvicki** [reply]
Will do.

p.s. Not a bro. :)

**Reply by GrampaWildWilly** [reply]
I am curious to see if not-bro gets results any different from those I posted upthread here.

**Reply by paulrouget** [reply]
@Niazi78675 this will work better with version 8.2.1 which should arrive very soon

**Reply by GrampaWildWilly** [reply]
How does 8.2.1 compare with 8.2.1.2a1 beta?  I just tried again with the new beta but got the same results.  Two variants just got an immediate error & another variant downloaded only 8 seconds of something.

**Reply by Niazi78675** [reply]
@paulrouget Any idea when is it coming? But still doesn't know whether it will also work or not

**Reply by paulrouget** [reply]
@GrampaWildWilly you are talking about @Niazi78675 issue?

**Reply by Niazi78675** [reply]
@GrampaWildWilly me too tried the beta version it also gave error  now waiting for @jazzinvicki to give it a try maybe has a solution

**Reply by Mattmjs** [reply]
I don't think it is possible the massage video can be downloaded like I said further up in a reply. Each stream manifest comes paired with
a key. This I believe is encryption. Your going to need the decryption key it if attempting to download. Which is impossible.
It will never work.

**Reply by GrampaWildWilly** [reply]
Turns out that's a red herring.  See my post from a few minutes ago upthread here.  GitHub has an annoying habit of hiding bunches of posts.  You'll probably have to click some of those "Show previous xx replies" buttons to expose it.

**Comment by Mattmjs** [comment]
Well well, so I was proven wrong on it.
I did think it was impossible, I don't know I would have thought to do all
that.
I guess you wanted a challenge.  👍
